packages/features/ee/round-robin/roundRobinReassignment.ts

```diff
@@ -0,0 +1,517 @@
+// eslint-disable-next-line no-restricted-imports
+import { cloneDeep } from "lodash";
+
+import { OrganizerDefaultConferencingAppType, getLocationValueForDB } from "@calcom/app-store/locations";
+import EventManager from "@calcom/core/EventManager";
+import { getEventName } from "@calcom/core/event";
+import dayjs from "@calcom/dayjs";
+import { sendRoundRobinCancelledEmails, sendRoundRobinScheduledEmails } from "@calcom/emails";
+import getBookingResponsesSchema from "@calcom/features/bookings/lib/getBookingResponsesSchema";
+import { getCalEventResponses } from "@calcom/features/bookings/lib/getCalEventResponses";
+import { ensureAvailableUsers } from "@calcom/features/bookings/lib/handleNewBooking/ensureAvailableUsers";
+import { getEventTypesFromDB } from "@calcom/features/bookings/lib/handleNewBooking/getEventTypesFromDB";
+import type { IsFixedAwareUser } from "@calcom/features/bookings/lib/handleNewBooking/types";
+import {
+  scheduleEmailReminder,
+  deleteScheduledEmailReminder,
+} from "@calcom/features/ee/workflows/lib/reminders/emailReminderManager";
+import { isPrismaObjOrUndefined } from "@calcom/lib";
+import logger from "@calcom/lib/logger";
+import { getLuckyUser } from "@calcom/lib/server";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import { BookingReferenceRepository } from "@calcom/lib/server/repository/bookingReference";
+import { getTimeFormatStringFromUserTimeFormat } from "@calcom/lib/timeFormat";
+import { prisma } from "@calcom/prisma";
+import { WorkflowActions, WorkflowMethods, WorkflowTriggerEvents } from "@calcom/prisma/enums";
+import { userMetadata as userMetadataSchema } from "@calcom/prisma/zod-utils";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+
+const bookingSelect = {
+  uid: true,
+  title: true,
+  startTime: true,
+  endTime: true,
+  userId: true,
+  customInputs: true,
+  responses: true,
+  description: true,
+  location: true,
+  eventTypeId: true,
+  destinationCalendar: true,
+  user: {
+    include: {
+      destinationCalendar: true,
+    },
+  },
+  attendees: true,
+  references: true,
+};
+
+export const roundRobinReassignment = async ({ bookingId }: { bookingId: number }) => {
```

Comment:
This function is really big and does a lot of things, I think we can improve that to make it better readable and easier to understand. This can be done later on though, I think we should focus on getting this merged now 🙏